src/update.ts

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+import * as uuid from 'uuid';
+import { bufferToHex } from 'ethereumjs-util'
+import { helpers } from 'leap-core';
+import Web3 from 'web3';
+
+import dynamoDb from './dynamodb'
+
+export const updateStats = async (_event, _context, callback) => {
+  const plasmaProvider = 'wss://testnet-node1.leapdao.org:1443';
+  const web3Plasma = helpers.extendWeb3(new Web3(plasmaProvider));
+
+  await web3Plasma.eth.net.getId();
+
+  const uTxos = await web3Plasma.getUnspent();
+
+  const unspents: import('./types').UnspentWithTx[] = await Promise.all(
+    uTxos.map((u: import('./types').UnspentWithTx) =>
+      web3Plasma.eth
+        .getTransaction(bufferToHex(u.outpoint.hash))
+        .then((tx: import('./types').LeapTransaction) => {
+          u.transaction = tx;
+          return u;
+        })
+        .then((uTx: import('./types').UnspentWithTx) =>
+          web3Plasma.eth
```

Comment:
We can cache blocks that we already requested with simple object, stored outside of this mapping. This will speed up the process and lower number of requests to node. Potentially the bill for using this function as well :)

Reply:
Not sure I get the idea of what you're suggesting here. You'd like me to save the unspent transactions in a DB table and just keep updating the new one's to process the stats?

I think my actual question is, wouldn't the old unspent transactions ever become spent/exit? I'm still trying to get my head around this 😓

Reply:
I was refering to block fetching. You are building an array of promises, and then waiting until all of the resolve, right? I suppose there can be cases, when we have several transactions in the same block, thus no need to refetch the block is we already done this before, right?

Reply:
Thanks for the pointer. I'll spend a bit more time thinking about this to see a way to optimize it right from the start 😅